numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -562,38 +562,24 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
         return ret;
     }
     else if (!PyBytes_Check(object) || PyBytes_GET_SIZE(object) < 1) {
-        /* 2015-12-14, 1.11 */
-        int ret = DEPRECATE("Non-string object detected for "
-                            "the array ordering. Please pass "
-                            "in 'C', 'F', 'A', or 'K' instead");
-
-        if (ret < 0) {
-            return -1;
-        }
-
-        if (PyObject_IsTrue(object)) {
-            *val = NPY_FORTRANORDER;
-        }
-        else {
-            *val = NPY_CORDER;
-        }
-        if (PyErr_Occurred()) {
-            return NPY_FAIL;
-        }
-        return NPY_SUCCEED;
+        /* 2015-12-14, 1.11 DEPRECATED */
+        /* 2019-09-25, 1.18 ERROR */
+        PyErr_SetString(PyExc_ValueError, "Non-string object detected for "
```

Comment:
I think for errors I prefer the formatting:
```
PyErr_SetString(PyExc_ValueError,
        "this is a very long error message .... ... ... .. "
        "...");
```
As for the above comment, I do not think we typically put that in, not that it hurts, though, hmmm. The idea of the DEPRECATED comment is to find these places that need updating easier. What is important here is only the release note.

Reply:
I updated the PyErr_SetString, I'm happy to
* remove the both dates
* consolidate them into a single line
* /* more order formats were handled before gh-14596 */

Reply:
Others may disagree, but if you do not mind, lets just remove the comments, they are not interesting internally for numpy, only for downstream. And downstream would read the release notes.

Can you reformat the other error setting as well?

The main thing to do is add the release note snippet, thanks!

Reply:
Removed comments and added a release note snippet.

Reply:
Thanks. LGTM. Do you want to squash it into a single commit with `DEP: ...`? I can also squash merge, but thought I would mention just in case, since we slightly prefer merges.

Reply:
I'm happy to do what you prefer.

I'm squashing and force committing a single commit with the correctish title